src/measures/fermions/ellesettete.cpp

```diff
@@ -101,27 +108,36 @@ namespace nissa
 		//loop over hits
 		for(int ihit=0;ihit<meas_pars.nhits;ihit++)
 		  {
-		    //fill the source
+		    //prepare the sources with the right structure spin x taste
 		    fill_source(source,glb_t,meas_pars.rnd_type);
-		    
-		    //compute std 2pts propagator G(m|n) ~ [D^-1(m|y) source(y)] source(n)*
+		    apply_stag_op(g5_id_source,conf,theory_pars.backfield[iflav],GAMMA_INT::GAMMA_5,GAMMA_INT::IDENTITY,source);
+		    apply_stag_op(id_g5_source,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,source);
+
+		    //compute std 2pts propagator G(m|n) ~ [D^-1(m|y) source(y)] source(n)* and simple sequential propagator
 		    MINV(SIMPLE_PROP,iflav,source);
-		    
-		    //compute sequential propagator G(m|n) ~ [D^-1(m|y) source(y)] D^-1(y|n)
 		    MINV(SEQ_PROP,iflav,SIMPLE_PROP);
-		    
-		    //then glb reduction to compute the trace for the connected 3pts diagram and the 2pts
-		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_two_pts,SIMPLE_PROP,SIMPLE_PROP);
-		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_three_pts,SIMPLE_PROP,SEQ_PROP);
-		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_four_pts,SEQ_PROP,SEQ_PROP);
 
-		    //////// disconnected ////////
-		    apply_stag_op(source_g5_id,conf,theory_pars.backfield[iflav],GAMMA_INT::GAMMA_5,GAMMA_INT::IDENTITY,source);
+		    //compute  2pts propagator with id x g5 at source and apply id x g5 at sink
+		    MINV(PROP_ID_G5,iflav,id_g5_source);
+		    apply_stag_op(ID_G5_PROP_ID_G5,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,PROP_ID_G5);
+
+		    //compute sequential propagator with id x g5 at source and apply id x g5 at sink
+		    MINV(SEQ_PROP_ID_G5,iflav,ID_G5_PROP);
```

Comment:
no... here you are putting the insertion of the identity over spin and g5 over taste, but the sequential must have identity over taste

Reply:
it should have been PROP_ID_G5 but I mispelled the name of the prop, which indeed doesn't exist in the code.